esda/moran.py

```diff
@@ -1700,12 +1743,76 @@ def by_col(
             outvals=outvals,
             swapname=swapname,
             stat=Moran_Local,  # how would this get done w/super?
-            **stat_kws
+            **stat_kws,
         )
         for col in rate_df.columns:
             df[col] = rate_df[col]
 
 
+def _explore_local_moran(moran_local, gdf, crit_value, explore_kwargs=None, m=None):
+    """Plot local Moran values as an interactive map
+
+    Parameters
+    ----------
+    moran_local : esda.Moran_Local
+        a fitted local Moran class from the PySAL esda module
+    gdf : geopandas.GeoDataFrame
+        geodataframe used to create the Moran_Local class
+    crit_value : float, optional
+        critical value for determining statistical significance, by default 0.05
+    explore_kwargs : dict, optional
+        additional keyword arguments passed to geopandas.explore, by default None
+    m : folium.Map, optional
+        folium map canvas to plot on top of, by default None
+
+    Returns
+    -------
+    m
+        folium.Map
+    """
+    if explore_kwargs is None:
+        explore_kwargs = dict()
+    gdf = gdf.copy()
+    gdf["Moran Cluster"] = moran_local.get_cluster_labels(crit_value)
+    gdf["p-value"] = moran_local.p_sim
+
+    x = gdf["Moran Cluster"].values
+    y = np.unique(x)
+    colors5_mpl = {
+        "High-High": "#d7191c",
+        "Low-High": "#abd9e9",
+        "Low-Low": "#2c7bb6",
+        "High-Low": "#fdae61",
+        "Insignificant": "lightgrey",
+    }
+    colors5 = [colors5_mpl[i] for i in y]  # for mpl
+    hmap = colors.ListedColormap(colors5)
```

Comment:
Not sure if this is bullet proof. See https://github.com/pysal/splot/issues/178. If you miss one of the clusters, this may get misaligned (haven't tested though).

Reply:
I saw that,  but I think this works. The test case has no "diamonds" but still renders correctly. I'm setting the colors a bit differently than stef did